src/Lecture3.hs

```diff
@@ -122,10 +132,14 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
-
+  (<>) :: Reward -> Reward -> Reward
+  Reward aGold aSpecial <> Reward bGold bSpecial = Reward { rewardGold    = aGold <> bGold
+                                                          , rewardSpecial = aSpecial || bSpecial }
 
 instance Monoid Reward where
-
+  mempty :: Reward
+  mempty = Reward { rewardGold    = Gold 0
```

Comment:
You can use `mempty` for `Gold` here because it already has a `Monoid` instance 🏅 
```suggestion
  mempty = Reward { rewardGold    = mempty
```